platform/src/main/java/org/stellar/anchor/platform/service/HealthCheckService.java

```diff
@@ -0,0 +1,42 @@
+package org.stellar.anchor.platform.service;
+
+import java.util.*;
+import org.springframework.stereotype.Service;
+import org.stellar.anchor.api.platform.HealthCheckResponse;
+import org.stellar.anchor.api.platform.HealthCheckResult;
+
+@Service
+public class HealthCheckService {
+  private final Map<String, List<HealthCheckable>> mapCheckable = new HashMap<>();
+
+  HealthCheckService(List<HealthCheckable> checkables) {
+    for (HealthCheckable checkable : checkables) {
+      for (String tag : checkable.getTags()) {
+        List<HealthCheckable> checksOfTag = mapCheckable.get(tag);
+        if (checksOfTag == null) {
+          checksOfTag = List.of(checkable);
+          mapCheckable.put(tag, checksOfTag);
+        } else {
+          checksOfTag.add(checkable);
+        }
```

Comment:
💡 this suggestion is just a personal preference of mine, so no need to make the change if you don't want:

```suggestion
        if (checksOfTag == null) {
          checksOfTag = List.of();
          mapCheckable.put(tag, checksOfTag);
        }
        checksOfTag.add(checkable);
```

Reply:
Good one. I used to do your way. But for some reason, I was doing the other way.
```java
        if (checksOfTag == null) {
          mapCheckable.put(tag, List.of());
        }
        checksOfTag.add(checkable);
```